tika-parsers/tika-parsers-standard/tika-parsers-standard-modules/tika-parser-html-module/src/main/java/org/apache/tika/parser/html/HtmlParser.java

```diff
@@ -146,7 +146,11 @@ metadata, getEncodingDetector(context))) {
             parser.setContentHandler(new XHTMLDowngradeHandler(
                     new HtmlHandler(mapper, handler, metadata, context, extractScripts)));
 
-            parser.parse(reader.asInputSource());
+            try {
+                parser.parse(reader.asInputSource());
+            } catch (StringIndexOutOfBoundsException e) {
```

Comment:
@tballison - any thoughts on Tika's general strategy for what errors get returned by parsers? E.g. is the trend towards always throwing a `TikaException`, versus more specific errors?